Mbedtls 2.28 tls improve

lategoodbye opened the pull request:
Improve ssl_parse_trusted_ca_keys_ext

During recent tests it was noticed that there are possibly issues
in parsing code.

At first move all TrustedAuthority length checks into the IdentifierType
switch. This should fix the behavior for pre_agreed and x509_name.

Second don't abort TLS handshake in error case instead ignore the
extension completely.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

lategoodbye committed:
Move debug Trusted CA size before type

This allows to return within the type switch.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>

FaHaGit commented on library/ssl_srv.c:
```diff
@@ -577,37 +577,59 @@ static int ssl_parse_trusted_ca_keys_ext( mbedtls_ssl_context *ssl,
+            {
+                MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad trusted_auth_size" ) );
+                ssl->handshake->trusted_ca_key_cert = NULL;
+                return ( 0 );
```
return( 0 );

FaHaGit replied:
space sign should be removed also in the other return() lines

lategoodbye replied:
Should be fixed now

FaHaGit reviewed (changes requested):
Only a little code style point. Everything else looks fine :)

mooraby commented on library/ssl_srv.c:
```diff
@@ -579,35 +579,57 @@ static int ssl_parse_trusted_ca_keys_ext( mbedtls_ssl_context *ssl,
+        if( trusted_auth_list_size + trusted_auth_size + 2 > len )
         {
-            MBEDTLS_SSL_DEBUG_MSG( 1, ( "bad client hello message" ) );
+#if 0
```
Can you add more explaination to this?

lategoodbye replied:
Please have a look at the new version

lategoodbye committed:
Improve ssl_parse_trusted_ca_keys_ext

During recent tests it was noticed that there are possibly issues
in parsing code.

At first move all TrustedAuthority length checks into the IdentifierType
switch. This should fix the behavior for pre_agreed and x509_name.

Second don't abort TLS handshake in error case instead ignore the
extension completely.

Signed-off-by: Stefan Wahren <stefan.wahren@chargebyte.com>
